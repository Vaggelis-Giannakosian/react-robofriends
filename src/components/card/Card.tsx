import React from 'react'
import {IRobot} from "../../containers/App/App";

const Card: React.FC<IRobot> = ({id, name, email}) => {

    return (
        <div className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
            <img src={`https://robohash.org/${id}?size=200x200`} alt="robot"/>
            <div>
                <h2>{name}</h2>
                <p>{email}</p>
            </div>
        </div>
    );

}

export default React.memo(Card);
