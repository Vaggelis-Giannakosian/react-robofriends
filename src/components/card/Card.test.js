import Card from './Card'
import {shallow, mount, render} from 'enzyme'


it('simple',()=>{
    expect.assertions(3)

    const robot = {
        id:1,
        name:'Vangelis',
        email:'e.giannakosian@gmail.com'
    }
    expect(4).toBe(4)
    expect(shallow(<Card/>).length).toEqual(1)
    expect(shallow(<Card id={robot.id} name={robot.name} email={robot.email} />)).toMatchSnapshot()
})