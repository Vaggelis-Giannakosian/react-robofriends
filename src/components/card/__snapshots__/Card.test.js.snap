// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple 1`] = `
<div
  className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robot"
    src="https://robohash.org/1?size=200x200"
  />
  <div>
    <h2>
      Vangelis
    </h2>
    <p>
      e.giannakosian@gmail.com
    </p>
  </div>
</div>
`;
